c7n/resources/eks.py

```diff
@@ -211,6 +211,72 @@ def process(self, resources):
                 "Filtered %d of %d clusters due to state", state_filtered, len(resources))
 
 
+@EKS.action_registry.register('associate-encryption-config')
+class AssociateEncryptionConfig(Action):
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['associate-encryption-config']},
+            'encryptionConfig': {
+                'type': 'array',
+                'properties': {
+                    'type': 'object',
+                    'properties': {
+                        'provider': {
+                            'type': 'object',
+                            'properties': {
+                                'keyArn': {'type': 'string'}
+                            }
+                        },
+                        'resources': {
+                            'type': 'array',
+                            'properties': {
+                                'enum': 'secrets'
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    permissions = ('eks:AssociateEncryptionConfig', 'kms:DescribeKey',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('eks')
+        state_filtered = 0
+        params = dict(self.data)
+        params.pop('type')
+        for r in resources:
+            if r['status'] != 'ACTIVE':
+                state_filtered += 1
+                continue
+            try:
+                client.associate_encryption_config(
+                    clusterName=r['name'],
+                    encryptionConfig=params['encryptionConfig']
+                )
+            except client.exceptions.InvalidParameterException:
+                # associate_encryption_config does not accept kms key aliases, if provided
+                # with an alias find the key arn with kms:DescribeKey.
```

Comment:
this feels like it can be proactively done, ie if the encryption config is an alias resolve it early, versus generating a failed api call for every single cluster.